### ISSUE TITLE
(Flatpak) (Solus Gnome) Issues with rendering window and more

### ISSUE THREAD

**Issue by RTSAjwad** [phase1geo/Minder#170]
I have the understanding that the flatpak isn't officially maintained after checking this github page. Regardless, I'd like to leave these issues and critiques here incase the maintainer sees it.

1. When dragging window around the first time, it is fine. Dragging a second time causes app to freeze for a few brief moments.
2. Going full screen causes the app to display visual glitches and freeze for a few brief moments. This only happens when dragging to the top to full screen. Using maximise/minimise buttons still displays glitches but it is much quicker.
3. Sidebar animations are slightly delayed.
4. Expanding sidebar options has no animations.
5. Resizing sidebar doesn't change cursor. Likewise hovering over text field doesn't change cursor.
6. I think it should use system gtk theme provided by flatpak. I had to manually set it myself with an environment variable. 
7. I'm guessing this one is to do with the main app itself but importing Freemind files exported from xmind8 doesn't retain the notes.

Apart from this, the app works fine. I'll just full screen it on its own workspace and it's still functional.

**Comment by phase1geo** [comment]
So #1, #2, #3, #6 are out of my hands.

#4 is not supported.  Opening the sidebar will not animate the open and closing the sidebar will not animate the close.  There is a technical reason for this, but I'd rather not get into it.

#5 and #7 are issues that I can and will address with this issue report.  In terms of #5, are you referring to the text field in this sidebar?  Or are you referring to text within a node in the mind map?

**Comment by phase1geo** [comment]
Supporting Flatpak currently is not a priority for the project development.  Other members of the Flatpak community have put together a Flatpak manifest for the project and have been updating it as we go along; however, I have not received pull requests from those contributors to fix Flatpak-specific issues.

Minder is primarily being developed for elementary OS which, in Hera, requires Debian packaging and sandboxing is not required/enforced.  That being said, as elementary OS moves to its next version (6.0 Odin), I believe the goal is to move to Flatpaks wholly.  If/when that happens, support for Flatpaks will be given due priority.

**Comment by phase1geo** [comment]
#7 has been taken care of in the devel branch.  This will be generally available in the next stable release.

**Comment by phase1geo** [comment]
Re-reading #6 this evening. I'm curious, what was the environment variable that you used to change the theme?

**Comment by RTSAjwad** [comment]
Sorry about the late replies. The environment variable was:
`GTK_THEME=Adwaita`
So altogether to run the application I use:
`GTK_THEME=Adwaita flatpak run com.github.phase1geo.minder`

**Comment by RTSAjwad** [comment]
> So #1, #2, #3, #6 are out of my hands.
> 
> #4 is not supported. Opening the sidebar will not animate the open and closing the sidebar will not animate the close. There is a technical reason for this, but I'd rather not get into it.
> 
> #5 and #7 are issues that I can and will address with this issue report. In terms of #5, are you referring to the text field in this sidebar? Or are you referring to text within a node in the mind map?

I was referring to the search bar when you click on the search button from the header bar.

**Comment by RTSAjwad** [comment]
> #7 has been taken care of in the devel branch. This will be generally available in the next stable release.

Thank you.

**Comment by RTSAjwad** [comment]
> Supporting Flatpak currently is not a priority for the project development. Other members of the Flatpak community have put together a Flatpak manifest for the project and have been updating it as we go along; however, I have not received pull requests from those contributors to fix Flatpak-specific issues.
> 
> Minder is primarily being developed for elementary OS which, in Hera, requires Debian packaging and sandboxing is not required/enforced. That being said, as elementary OS moves to its next version (6.0 Odin), I believe the goal is to move to Flatpaks wholly. If/when that happens, support for Flatpaks will be given due priority.

Yes, I understand. I realised after reading through this repository. I didn't really know where I could report the issues so I just put flatpak in my title :/

If a move to flatpaks was to happen then that would be cool. Probably be easier to address non elementary specific issues too. I used to use elementary and it was good :)
